app/controllers/system_admin/providers_controller.rb

```diff
@@ -43,6 +43,15 @@ def update
       end
     end
 
+    def destroy
+      @provider.users.delete_all
+
+      @provider.discard!
```

Comment:
Should this be the other way round? If it raises, we would have deleted the users already

Reply:
fixed